hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/utils/MiscCryptoUtils.java

```diff
@@ -62,11 +97,23 @@ public static byte[] decompressSecp256k1(final byte[] compressedKey) {
      * @return the raw bytes of the compressed public key
      */
     public static byte[] compressSecp256k1(final byte[] decompressedKey) {
-        final var decompressedBytes = new byte[65];
-        decompressedBytes[0] = 0x04;
-        System.arraycopy(decompressedKey, 0, decompressedBytes, 1, 32);
-        System.arraycopy(decompressedKey, 32, decompressedBytes, 33, 32);
-        return curveSecp256k1.decodePoint(decompressedBytes).getEncoded(true);
+        final ThreadLocalCache cache = CACHE.get();
+        // add header byte on to decompressed key
+        final byte[] decompressedBytes = cache.uncompressedPublicKeyInput;
+        System.arraycopy(decompressedKey, 0, decompressedBytes, 1, ECDSA_UNCOMPRESSED_KEY_SIZE);
+        // convert public key to native format
+        final LibSecp256k1.secp256k1_pubkey publicKey = cache.pubKey;
+        final int keyParseResult = LibSecp256k1.secp256k1_ec_pubkey_parse(
+                LibSecp256k1.CONTEXT, publicKey, decompressedBytes, decompressedBytes.length);
+        if (keyParseResult != 1) throw new RuntimeException("Failed to parse public key");
```

Comment:
I wonder if we can use a more specific exception for this case and the serialization case below. Perhaps a CryptographyException, ParseException, or something similar?
Using generic RuntimeException throws away all information about what failed, so some top-level handler just runs a generic "unknown failure" path instead of handling the parse or serialization problem properly.

Reply:
Thanks, throwing IllegalArgumentException as that is what old code did and what is handled in modular services code